File: packages/web-client/src/types/global-options.ts


class GlobalOptions {

    private static instance: GlobalOptions;
    baseUrl: string;

    private constructor(baseUrl: string) {
        this.baseUrl = baseUrl;
    }


    public static getInstance(baseUrl: string): GlobalOptions {


        if (!GlobalOptions.instance) {
            GlobalOptions.instance = new GlobalOptions(baseUrl);
        }

        return GlobalOptions.instance;
    }
}

export default GlobalOptions;

File: packages/flutter/lib/src/class/global_options.dart
class GlobalOptions {
  String _baseURL;
  String get baseUrl => _baseURL;

  static GlobalOptions _instance = GlobalOptions._internal();

  GlobalOptions._internal();

  factory GlobalOptions() {
    return _instance;
  }

  void setOption({String baseUrl}) {
    this._baseURL = baseUrl;
  }
}
